doc: add private packages usage documentation (#153)
gatling/frontline-cloud-doc: content/reference/admin/private_locations/private_packages/index.md

---
title: "Private Packages"
description: "How to install and use a private repository on your Control Plane"

> lead: "Keep your simulations packages privately in your infrastructure, and use them along private locations"

Reviewer: ```suggestion
lead: "Store your simulations' packages privately in your infrastructure, and use them with private locations"
```

date: 2023-08-09T12:00:00+00:00
lastmod: 2023-08-09T12:00:00+00:00
weight: 22033
---

# Introduction

> Using private locations allows you to run your simulations in your private cloud. Now, what about your simulations packages ? They may hold private information, so you may want to store them in your private cloud also.

Reviewer: ```suggestion
Using Private Locations allows you to run your simulations in your private cloud. Now, what about your simulations' packages? They may hold private information, so you may want to also store them in your private cloud.
```

> Your control plane has a private repository feature, allowing you to manage your simulation packages privately, you just have to enable it.

Reviewer: ```suggestion
Your Control Plane has a private repository feature, allowing you to manage your simulation packages privately, you just have to enable it.
```

# Private packages

## Infrastructure

{{< img src="private_packages_general_architecture.png" alt="Infrastructure schema" >}}

{{< alert info >}}

> Private packages are based on private locations. You can not use them with managed locations.

Reviewer: ```suggestion
Private Packages are based on Private Locations. You can not use them with managed locations.
```

{{< /alert >}}

## Installation

First step: follow the control plane installation guide for private locations.

{{< alert info >}}
Before going further, ensure that your AWS S3 bucket is ready to hold your packages.
{{< /alert >}}

{{< alert warning >}}
Control plane with private repository needs AWS permissions `s3:PutObject`` and `s3:DeleteObject` on the bucket.
{{< /alert >}}

> Once it is done, just add private repository configuration section in your [control plane configuration]({{< ref "../introduction" >}})  file :

Reviewer: ```suggestion
Once it is done, add the private repository configuration section in your [control plane configuration]({{< ref "../introduction" >}})  file :
```

```bash
control-plane {

>   # ... existing configuration

Reviewer 2: Yes. remove this line.
